plugins/provisioners/puppet/config/puppet.rb

```diff
@@ -95,10 +102,15 @@ def finalize!
             @environment_variables = {}
           end
 
+          if @synced_folder_opts.eql?({})
+            @synced_folder_opts[:type] = @synced_folder_type if @synced_folder_type
+            @synced_folder_opts[:owner] = "root" if !@synced_folder_type
+            @synced_folder_opts[:args] = @synced_folder_args if @synced_folder_args
+            @synced_folder_opts[:nfs__quiet] = true
+          end
+
           @binary_path        = nil     if @binary_path == UNSET_VALUE
           @module_path        = nil     if @module_path == UNSET_VALUE
-          @synced_folder_type = nil     if @synced_folder_type == UNSET_VALUE
-          @synced_folder_args = nil if @synced_folder_args == UNSET_VALUE
```

Comment:
This should still be here I think. Otherwise these objects will be left as `UNSET_VALUE` internally and could lead to some exceptions down the road.

Reply:
Right now this will not be the case, because i removed the ```@synced_folder_type``` from above at line 38. And the ```@synced_folder_args``` was not present there anyway, and i guess they were present at this place to be able to access them from ```@config.synced_folder_*``` within ```plugins/provisioners/puppet/provisioner/puppet.rb```. 
And since i moved that piece of code into the config we don't need them here any longer. 
Or do i misunderstand the ruby language here?

Shouldn't it be sufficient to re-add lines 20 and 21 above and then access them within between lines 105 - 110 above?

Reply:
Sure, thanks :) Now i am able to follow your reasoning.